modules/kernel/src/main/java/com/thegoate/expect/builder/ExpectationBuilder.java

```diff
@@ -1,141 +1,153 @@
-            if (!("" + actual).contains("*") && !("" + actual).contains("+")) {
-                a = new Get(actual).from(from);
-            } else {
-                a = from;
-            }
-        }
-        return a;
-    }
-
-    public ExpectationBuilder setActual(Object actual) {
-        if(actualValue == null){
-            actualValue = new Value(actual);
-        }
-        this.actual = actual;
-        return this;
-    }
-
-    public Object getExpected() {
-        return expected;
-    }
-
-    public ExpectationBuilder setExpected(Object expected) {
-        if(expectedValue == null){
-            expectedValue = new Value(expected);
-        }
-        this.expected = expected;
-        return this;
-    }
-}
+/*
+ * Copyright (c) 2017. Eric Angeli
+ *
+ *  Permission is hereby granted, free of charge,
+ *  to any person obtaining a copy of this software
+ *  and associated documentation files (the "Software"),
+ *  to deal in the Software without restriction,
+ *  including without limitation the rights to use, copy,
+ *  modify, merge, publish, distribute, sublicense,
+ *  and/or sell copies of the Software, and to permit
+ *  persons to whom the Software is furnished to do so,
+ *  subject to the following conditions:
+ *
+ *  The above copyright notice and this permission
+ *  notice shall be included in all copies or substantial
+ *  portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ *  WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE
+ *  AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *  WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ *  DEALINGS IN THE SOFTWARE.
+ */
+package com.thegoate.expect.builder;
+
+import com.thegoate.expect.Expectation;
+import com.thegoate.utils.get.Get;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Eric Angeli on 3/26/2019.
+ */
+public abstract class ExpectationBuilder {
+    protected Object from;
+    protected Object fromExpected;
+    protected Object actual;
+    protected Object expected;
+    protected List<Expectation> expectations = new ArrayList<>();
+    protected Value actualValue;
+    protected Value expectedValue;
+    protected long period = 50;
+    protected long timeoutMS = 0;
+//    public List<Expectation> build(Expectation expectation){
+//        Goate exp = (Goate)expectation.getExpectations().get(0);
+//        setFrom(expectation.getFrom());
+//        setActual(exp.get("actual"));
+//        return build();
+//    }
+
+    public abstract List<Expectation> build();
+
+    protected ExpectationBuilder expect(Expectation expectation) {
+        expectations.add(expectation.retryTimeout(timeoutMS).retryPeriod(period));
```

Comment:
the framework uses a bad hack to determine if the timeout and period was set explicitly for the individual expectation. always setting it would violate that hack. Two choices are to initialize the timeout and period to a value <0, ie: -42. Or to have a check to see if the values have been set and only set them if they have been.

Reply:
These values are initialized on lines 46-47
```java
protected long period = 50;
protected long timeoutMS = 0;